java/com/google/re2j/Pattern.java

```diff
@@ -269,6 +270,14 @@ public int groupCount() {
     return re2.numberOfCapturingGroups();
   }
 
+  /**
+   * Return a map of the capturing groups in this matcher's pattern, where key is the name and value
+   * is the index of the group in the pattern.
+   */
+  public Map<String, Integer> namedGroups() {
+    return re2.namedGroups;
```

Comment:
should we defensively copy the map to ensure that callers don't change it?

Reply:
done.